data-access/mandataris-download.ts

```diff
@@ -0,0 +1,261 @@
+}
+
+function createFractieFilter(filters): string {
+  const {
+    fractieIds,
+    hasFilterOnOnafhankelijkeFractie,
+    hasFilterOnNietBeschikbareFractie,
+  } = filters;
+
+  if (
+    fractieIds.length === 0 &&
+    !hasFilterOnNietBeschikbareFractie &&
+    !hasFilterOnOnafhankelijkeFractie
+  ) {
+    return '';
+  }
+
+  const idValues = fractieIds.map((id) => sparqlEscapeString(id)).join('\n');
+  const unions = [
+    {
+      apply: fractieIds.length >= 1,
+      filter: `
+        UNION {
+          VALUES ?fractieId {\n ${idValues} }
+          ?fractie mu:uuid ?fractieId.
+        }
+      `,
+    },
+    {
+      apply: hasFilterOnNietBeschikbareFractie,
+      filter: `
+        UNION {
+         FILTER NOT EXISTS {
+           ?mandataris org:hasMembership ?lidmaatschap.
+           ?lidmaatschap org:organisation ?fractie.
+         }
+        }
+      `,
+    },
+    {
+      apply: hasFilterOnOnafhankelijkeFractie,
+      filter: `
+        UNION {
+         ?fractie ext:isFractietype fractieType:Onafhankelijk.
+        }
+      `,
+    },
+  ];
+  const unionFilters = unions
+    .filter((union) => union.apply)
+    .map((union) => union.filter)
+    .join('\n');
+
+  return `
+    {
+      ?mandataris org:hasMembership ?lidmaatschap.
+      ?lidmaatschap org:organisation ?fractie.
+    }${unionFilters}
+  `;
+}
+
+async function getPropertiesOfMandatarissen(
+  mandatarisUris: Array<string>,
+  bestuursorgaanInTijdId: string | null,
+  sort: { ascOrDesc: 'ASC' | 'DESC'; filterProperty: string } | null,
+): Promise<Array<{ [key: string]: string }>> {
+  let sortFilter: string | null = null;
+  let bestuursorgaanInTijdFilter: string | null = null;
+
+  if (sort) {
+    sortFilter = `
+      ORDER BY ${sort.ascOrDesc}(${sort.filterProperty})
+    `;
+  }
+
+  if (bestuursorgaanInTijdId) {
+    bestuursorgaanInTijdFilter = `?bestuursorgaanInTijd mu:uuid ${sparqlEscapeString(
+      bestuursorgaanInTijdId,
+    )}.`;
+  }
+
+  const escapedUriValues = mandatarisUris
+    .map((uri) => sparqlEscapeUri(uri))
+    .join('\n');
+  const queryString = `
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX lmb: <http://lblod.data.gift/vocabularies/lmb/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+    PREFIX persoon: <http://data.vlaanderen.be/ns/persoon#>
+    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+    PREFIX regorg: <https://www.w3.org/ns/regorg#>
+
+    SELECT 
+      ?Voornaam
+      ?Naam 
+      (?fractieLabel as ?Fractie)
+      (?mandaatLabel as ?Mandaat)
+      (?statusLabel as ?Status)
+      (?bestuursorgaanLabel as ?Orgaan)
+      (?start as ?StartMandaat)
+      (?einde as ?EindeMandaat)
+      (?publicatieStatusLabel as ?PublicatieStatus)
+      ?Rangorde
+      (GROUP_CONCAT(DISTINCT ?beleidsdomeinLabel; SEPARATOR=" / ") AS ?Beleidsdomeinen)
```

Comment:
Only changed the variable name when it was conflicting to use the variable in the query. What I mean e.g. `?fractie` and `?Fractie`